### ISSUE TITLE
blocks-concurrent-subscriber stops

### ISSUE THREAD

**Issue by ko1nksm** [groovenauts/blocks-concurrent-subscriber#16]
I using blocks-concurrent-subscriber 0.2.1.
It stops sometimes in a few minutes. Restarting blocks-concurrent-subscriber is required to continue.

my configurationa file.

```
{
  "datasource": "root:@/DATABASE?parseTime=true",
  "agent": {
    "root-url": "http://concurrent-batch-agent-dot-PROJECT.appspot.com",
    "token": "{{ .TOKEN }}"
  },
  "subscriptions": [],
  "message-per-pull": 100,
  "interval": 10,
  "log-level": "debug",
  "patterns": [
    {
      "completed": "true",
      "command": ["./recv.sh", "COMPLETED", "%{job_message_id}"]
    },
    {
      "level": "fatal",
      "completed": "false",
      "command": ["./recv.sh", "FATAL", "job_message_id: %{job_message_id}, msg: %{data}"]
    },
    {
      "level": "error",
      "command": ["./recv.sh", "ERROR", "job_message_id: %{job_message_id}, msg: %{data}"]
    }
  ]
}
```

**recv.sh**

```
#!/bin/bash

echo $*
```

**logs**

```
./blocks-concurrent-subscriber_darwin_amd64 -c blocks-concurrent-subscriber.json
INFO[0000] Connecting to mysql database by root:@/DATABASE?parseTime=true

DEBU[0000] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0000] Pulling                                       subscriptions=[]
DEBU[0011] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0011] Pulling                                       subscriptions=[]
DEBU[0021] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0021] Pulling                                       subscriptions=[]
DEBU[0032] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0032] Pulling                                       subscriptions=[]
DEBU[0042] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0042] Pulling                                       subscriptions=[]
DEBU[0053] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0053] Pulling                                       subscriptions=[]
DEBU[0063] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0063] Pulling                                       subscriptions=[]
DEBU[0074] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0074] Pulling                                       subscriptions=[]
DEBU[0084] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0084] Pulling                                       subscriptions=[]
DEBU[0095] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0095] Pulling                                       subscriptions=[]
DEBU[0105] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0105] Pulling                                       subscriptions=[]
DEBU[0116] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0116] Pulling                                       subscriptions=[]
DEBU[0126] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0126] Pulling                                       subscriptions=[]
DEBU[0137] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0137] Pulling                                       subscriptions=[]
DEBU[0147] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0147] Pulling                                       subscriptions=[]
DEBU[0158] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0158] Pulling                                       subscriptions=[]
DEBU[0168] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0168] Pulling                                       subscriptions=[]
DEBU[0179] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0179] Pulling                                       subscriptions=[]
DEBU[0189] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0189] Pulling                                       subscriptions=[]
DEBU[0200] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0200] Pulling                                       subscriptions=[]
DEBU[0210] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0210] Pulling                                       subscriptions=[]
DEBU[0221] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0221] Pulling                                       subscriptions=[]
DEBU[0232] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0232] Pulling                                       subscriptions=[]
DEBU[0242] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0242] Pulling                                       subscriptions=[]
DEBU[0253] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0253] Pulling                                       subscriptions=[]
DEBU[0264] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0264] Pulling                                       subscriptions=[]
DEBU[0274] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0274] Pulling                                       subscriptions=[]
DEBU[0285] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0285] Pulling                                       subscriptions=[]
DEBU[0295] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0295] Pulling                                       subscriptions=[]
DEBU[0306] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0306] Pulling                                       subscriptions=[]
DEBU[0316] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0316] Pulling                                       subscriptions=[]
DEBU[0327] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0327] Pulling                                       subscriptions=[]
DEBU[0337] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0337] Pulling                                       subscriptions=[]
DEBU[0348] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0348] Pulling                                       subscriptions=[]
DEBU[0358] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0358] Pulling                                       subscriptions=[]
DEBU[0369] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0369] Pulling                                       subscriptions=[]
DEBU[0379] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0379] Pulling                                       subscriptions=[]
DEBU[0390] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0390] Pulling                                       subscriptions=[]
DEBU[0400] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0400] Pulling                                       subscriptions=[]
DEBU[0411] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0411] Pulling                                       subscriptions=[]
DEBU[0421] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0421] Pulling                                       subscriptions=[]
DEBU[0432] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0432] Pulling                                       subscriptions=[]
DEBU[0442] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0442] Pulling                                       subscriptions=[]
DEBU[0453] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0453] Pulling                                       subscriptions=[]
DEBU[0463] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0463] Pulling                                       subscriptions=[]
DEBU[0474] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0474] Pulling                                       subscriptions=[]
DEBU[0484] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0484] Pulling                                       subscriptions=[]
DEBU[0495] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0495] Pulling                                       subscriptions=[]
DEBU[0506] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0506] Pulling                                       subscriptions=[]
DEBU[0516] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0516] Pulling                                       subscriptions=[]
DEBU[0527] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0527] Pulling                                       subscriptions=[]
DEBU[0537] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0537] Pulling                                       subscriptions=[]
DEBU[0548] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0548] Pulling                                       subscriptions=[]
DEBU[0558] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0558] Pulling                                       subscriptions=[]
DEBU[0569] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0569] Pulling                                       subscriptions=[]
DEBU[0580] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0580] Pulling                                       subscriptions=[]
DEBU[0590] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0590] Pulling                                       subscriptions=[]
DEBU[0601] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0601] Pulling                                       subscriptions=[]
DEBU[0611] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0611] Pulling                                       subscriptions=[]
DEBU[0621] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0621] Pulling                                       subscriptions=[]
DEBU[0632] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0632] Pulling                                       subscriptions=[]
DEBU[0642] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0642] Pulling                                       subscriptions=[]
DEBU[0653] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0653] Pulling                                       subscriptions=[]
DEBU[0663] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0663] Pulling                                       subscriptions=[]
DEBU[0674] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0674] Pulling                                       subscriptions=[]
DEBU[0684] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0684] Pulling                                       subscriptions=[]
DEBU[0695] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0695] Pulling                                       subscriptions=[]
DEBU[0705] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0705] Pulling                                       subscriptions=[]
DEBU[0716] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0716] Pulling                                       subscriptions=[]
DEBU[0726] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0726] Pulling                                       subscriptions=[]
DEBU[0737] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0737] Pulling                                       subscriptions=[]
DEBU[0747] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0747] Pulling                                       subscriptions=[]
DEBU[0758] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0758] Pulling                                       subscriptions=[]
DEBU[0768] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0768] Pulling                                       subscriptions=[]
DEBU[0779] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0779] Pulling                                       subscriptions=[]
DEBU[0789] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0789] Pulling                                       subscriptions=[]
DEBU[0800] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0800] Pulling                                       subscriptions=[]
DEBU[0810] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0810] Pulling                                       subscriptions=[]
DEBU[0821] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0821] Pulling                                       subscriptions=[0xc820211f80]
DEBU[0897] Message received                              completed=false data="INITIALIZING STARTING" job_message_id=127806094041559 level=info progress=1
DEBU[0897] Message parsed                                completed=false data="INITIALIZING STARTING" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.782 +0000 UTC
DEBU[0897] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="INITIALIZING STARTING" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.782 +0000 UTC
DEBU[0897] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="INITIALIZING STARTING" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.782 +0000 UTC
DEBU[0897] Matched patterns                              matched=[] msg=&{127806094041559 1 {63631973184 782000000 0x95a180} false info INITIALIZING STARTING}
DEBU[0897] Commit
DEBU[0898] Message received                              completed=false data="INITIALIZING SUCCESS" job_message_id=127806094041559 level=info progress=1
DEBU[0898] Message parsed                                completed=false data="INITIALIZING SUCCESS" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.817 +0000 UTC
DEBU[0898] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="INITIALIZING SUCCESS" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.817 +0000 UTC
DEBU[0898] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="INITIALIZING SUCCESS" job_message_id=127806094041559 level=info progress=1 publishTime=2017-06-02 04:06:24.817 +0000 UTC
DEBU[0898] Matched patterns                              matched=[] msg=&{127806094041559 1 {63631973184 817000000 0x95a180} false info INITIALIZING SUCCESS}
DEBU[0898] Commit
DEBU[0898] Message received                              completed=false data="DOWNLOADING STARTING" job_message_id=127806094041559 level=info progress=2
DEBU[0898] Message parsed                                completed=false data="DOWNLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:06:24.834 +0000 UTC
DEBU[0898] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="DOWNLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:06:24.834 +0000 UTC
DEBU[0898] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="DOWNLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:06:24.834 +0000 UTC
DEBU[0898] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973184 834000000 0x95a180} false info DOWNLOADING STARTING}
DEBU[0898] Commit
DEBU[0909] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0909] Pulling                                       subscriptions=[0xc8202105c0]
DEBU[0951] Message received                              completed=false data="EXECUTING STARTING" job_message_id=127806094041559 level=info progress=2
DEBU[0951] Message parsed                                completed=false data="EXECUTING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.569 +0000 UTC
DEBU[0951] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="EXECUTING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.569 +0000 UTC
DEBU[0951] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="EXECUTING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.569 +0000 UTC
DEBU[0951] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973238 569000000 0x95a180} false info EXECUTING STARTING}
DEBU[0951] Commit
DEBU[0952] Message received                              completed=false data="DOWNLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2
DEBU[0952] Message parsed                                completed=false data="DOWNLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.551 +0000 UTC
DEBU[0952] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="DOWNLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.551 +0000 UTC
DEBU[0952] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="DOWNLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:07:18.551 +0000 UTC
DEBU[0952] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973238 551000000 0x95a180} false info DOWNLOADING SUCCESS}
DEBU[0952] Commit
DEBU[0962] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[0962] Pulling                                       subscriptions=[0xc820266340]
DEBU[1061] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1061] Pulling                                       subscriptions=[0xc820210780]
DEBU[1161] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1161] Pulling                                       subscriptions=[0xc820210340]
DEBU[1201] Message received                              completed=false data="EXECUTING SUCCESS" job_message_id=127806094041559 level=info progress=2
DEBU[1201] Message parsed                                completed=false data="EXECUTING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.622 +0000 UTC
DEBU[1201] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="EXECUTING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.622 +0000 UTC
DEBU[1201] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="EXECUTING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.622 +0000 UTC
DEBU[1201] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973489 622000000 0x95a180} false info EXECUTING SUCCESS}
DEBU[1201] Commit
DEBU[1202] Message received                              completed=false data="UPLOADING STARTING" job_message_id=127806094041559 level=info progress=2
DEBU[1202] Message parsed                                completed=false data="UPLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.644 +0000 UTC
DEBU[1202] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="UPLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.644 +0000 UTC
DEBU[1202] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="UPLOADING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:11:29.644 +0000 UTC
DEBU[1202] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973489 644000000 0x95a180} false info UPLOADING STARTING}
DEBU[1202] Commit
DEBU[1212] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1212] Pulling                                       subscriptions=[0xc8202663c0]
DEBU[1232] Message received                              completed=false data="UPLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2
DEBU[1232] Message parsed                                completed=false data="UPLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.04 +0000 UTC
DEBU[1232] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="UPLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.04 +0000 UTC
DEBU[1232] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="UPLOADING SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.04 +0000 UTC
DEBU[1232] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973520 40000000 0x95a180} false info UPLOADING SUCCESS}
DEBU[1232] Commit
DEBU[1232] Message received                              completed=false data="ACKSENDING STARTING" job_message_id=127806094041559 level=info progress=2
DEBU[1232] Message parsed                                completed=false data="ACKSENDING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.058 +0000 UTC
DEBU[1232] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="ACKSENDING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.058 +0000 UTC
DEBU[1232] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="ACKSENDING STARTING" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.058 +0000 UTC
DEBU[1232] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973520 58000000 0x95a180} false info ACKSENDING STARTING}
DEBU[1232] Commit
DEBU[1243] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1243] Pulling                                       subscriptions=[0xc82024e380]
DEBU[1244] Message received                              completed=false data="CLEANUP SUCCESS" job_message_id=127806094041559 level=info progress=2
DEBU[1244] Message parsed                                completed=false data="CLEANUP SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.872 +0000 UTC
DEBU[1244] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=false data="CLEANUP SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.872 +0000 UTC
DEBU[1244] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=false data="CLEANUP SUCCESS" job_message_id=127806094041559 level=info progress=2 publishTime=2017-06-02 04:12:00.872 +0000 UTC
DEBU[1244] Matched patterns                              matched=[] msg=&{127806094041559 2 {63631973520 872000000 0x95a180} false info CLEANUP SUCCESS}
DEBU[1244] Commit
DEBU[1254] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1254] Pulling                                       subscriptions=[0xc8202662c0]
DEBU[1255] Message received                              completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5
DEBU[1255] Message parsed                                completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC
DEBU[1255] Update pipeline_jobs successfully             SQL="UPDATE pipeline_jobs SET progress = ?, updated_at = ? WHERE job_message_id = ? AND progress < ?" completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC
DEBU[1255] Insert into pipeline_job_logs successfully    SQL="INSERT INTO pipeline_job_logs (pipeline, job_message_id, publish_time, progress, completed, log_level, log_message) VALUES (?, ?, ?, ?, ?, ?, ?)" completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC
DEBU[1255] Matched patterns                              matched=[0xc8200155e0] msg=&{127806094041559 5 {63631973520 782000000 0x95a180} true info ACKSENDING SUCCESS}
DEBU[1255] Executing command                             pattern=&{true   [./recv.sh COMPLETED %{job_message_id}]}
DEBU[1255] Expanding variables                           completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC template="./recv.sh"
DEBU[1255] Expanding variables                           completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC template=COMPLETED
DEBU[1255] Expanding variables                           completed=true data="ACKSENDING SUCCESS" job_message_id=127806094041559 level=info progress=5 publishTime=2017-06-02 04:12:00.782 +0000 UTC template="%{job_message_id}"
DEBU[1255] Executing command                             cmd=&{./recv.sh [./recv.sh COMPLETED 127806094041559] []  <nil> 0xc82002a010 0xc82002a018 [] <nil> <nil> <nil> <nil> false [] [] [] [] <nil>} pattern=&{true   [./recv.sh COMPLETED %{job_message_id}]}
COMPLETED 127806094041559
DEBU[1255] Commit
DEBU[1266] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1266] Pulling                                       subscriptions=[]
DEBU[1276] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1276] Pulling                                       subscriptions=[]
DEBU[1287] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1287] Pulling                                       subscriptions=[]
DEBU[1297] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1297] Pulling                                       subscriptions=[]
DEBU[1308] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1308] Pulling                                       subscriptions=[]
DEBU[1318] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1318] Pulling                                       subscriptions=[]
DEBU[1329] GET OK                                        status=200 url="http://concurrent-batch-agent-dot-PROJECT.appspot.com/pipelines/subscriptions"
DEBU[1329] Pulling                                       subscriptions=[]
(It seems stopped. I've waited over 15 minutes)
```

**Comment by ko1nksm** [comment]
I executed it after deleting GAE's task queue and data store, but it did not solve it.

**Comment by ko1nksm** [comment]
Issue was fixed in 0.2.3. thanks.
